discord/client.py

```diff
@@ -1473,11 +1474,14 @@ async def fetch_channel(self, channel_id: int) -> Union[GuildChannel, PrivateCha
             raise InvalidData('Unknown channel type {type} for channel ID {id}.'.format_map(data))
 
         if ch_type in (ChannelType.group, ChannelType.private):
-            channel = factory(me=self.user, data=data, state=self._connection)
+            # the factory will be a DMChannel or GroupChannel here
+            channel = factory(me=self.user, data=data, state=self._connection) # type: ignore
```

Comment:
I'm not sure what is being ignored. Isn't `factory` typed as `Any`?

Reply:
The (inferred, since the return type for factory related methods in channel.py isn't typed yet) type of  `factory` is `"Type[Thread] | Type[DMChannel] | Type[GroupChannel] | Type[TextChannel] | Type[VoiceChannel] | Type[CategoryChannel] | Type[StoreChannel] | Type[StageChannel] |`

DMChannel and GroupChannel are the only types of those that would take the `me` parameter.

Reply:
`mypy` doesn't infer return types, so I'm not used to `factory` being anything but `Any`. If `factory` is getting a type, then you definitely need to ignore when calling it.